modules/benchmark/R/calc.benchmark.R

```diff
@@ -18,29 +18,33 @@ calc.benchmark <- function(settings, con){ #settings file is output from start.b
   
   # Query BRR for benchmark ids
   
-  bm.id <- settings.bm$run$id #maybe something like that?
+  bm.id <- settings$benchmark$id 
+  obvs.id <- settings$benchmark$input_id
   
-  # For each bm.id
-  #  Query database for: input path, variables, site, start/end dates, metrics, formats
   
-  input.path <- db.query.file.path(input.id,host_name,con)
-  f_v <- db.query.format.vars(input.id,con)
-  format_table <- f_v[[1]]
-  vars_names_units <- f_v[[2]]
+  # For each bm.id - don't have this written as a loop yet. 
+  #  Query database for: input path, variables, site, start/end dates, metrics, formats
   
+  data.path <- query.file.path(obvs.id,settings$run$host$name,con)
+  format <- query.format.vars(obvs.id,con)  
   
-  # Need to do something here to narrow down the variables to be used
-  # rows <- which(vars_names_units$bety_name %in% calc.vars)
-  # vars_names_units <- vars_names_units[rows,]
+  site  <- query.site(settings$run$site$id, con)
   
-  site  <- db.query.site(input.id, con)
+  metrics <- db.query(paste("SELECT m.name from metrics as m JOIN benchmarks_metrics as b 
+                              ON m.id = b.metric_id WHERE b.benchmark_id = ", bm.id),con)
   
-  metrics <- db.query(paste("SELECT m.id, m.name from metrics as m JOIN benchmarks_metrics as b 
-                            ON m.id = b.metric_id WHERE b.benchmark_id = ", bm.id),con)
+  model_run <- dir(settings$run$host$outdir, full.names=TRUE)[1]
+  #How are we dealing with ensemble runs? Right now I've hardcoded to select the first run.
   
   # Local or remote
-  results <- calc.metrics(input_path, format_table, vars_names_units, model_run, metrics, 
-                          start_year=NA, end_year=NA, site=NA)
+  # format is a list, metrics is a dataframe, both can be long. How to pass to calc.metrics remotely? 
+  # Maybe as an Rdata file?
+  
+  start_year <- year(settings$run$start.date)
```

Comment:
@bcow you should go through the entirety of your code an rethink what will and will not be in settings.  I anticipate that the settings for running a whole suite of benchmarks will include host information, but not any run information since you're doing a standard run based on settings in the database, not in the settings file.